backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

```diff
@@ -57,15 +57,16 @@ protected void post(HttpServletRequest request, HttpServletResponse response) th
     String ingredients = request.getReader().readLine().replaceAll(" ", "%20");
     Entity userEntity = DatastoreUtils.getUserEntity();
     List<String> diets = DatastoreUtils.getPropertyAsList(userEntity, UserConstants.PROPERTY_DIETS);
-    List<String> allergies =
-        DatastoreUtils.getPropertyAsList(userEntity, UserConstants.PROPERTY_ALLERGIES);
+    ArrayList<String> allergies = new ArrayList<>
```

Comment:
You can still declare this as `List<String>` but initialize it as an `ArrayList`. You also shouldn't need to mark parameters as ArrayList (do the methods not work if you just expect a List?)

Reply:
Collections.addAll() was throwing an UnsupportedOperationException, but now that I searched up online again, I found out it was due to the Collections.emptyList() being an immutable list. I will change the declaration back to List